### ISSUE TITLE
MD037: False positive on underscores in filenames/identifiers

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#276]
## Summary

MD037 (no-space-in-emphasis) incorrectly flags underscores in filenames and identifiers as emphasis markers with spaces.

## Example

```markdown
_hello_world.rs_ rather than _helloworld.rs_.
```

We flag `_hello_world.rs_` as having "spaces inside emphasis markers" but this is a filename with underscores, not emphasis.

## Other Examples

- `_src/garden.rs_` - italicized path
- Variable names with underscores in italics

## Impact

| Linter | MD037 Violations |
|--------|------------------|
| mdbook-lint | 12 |
| rumdl | 0 |
| markdownlint | 0 |

## Root Cause

The rule detects `_word_with_underscores_` and thinks the underscores between words are emphasis markers with spaces. It should recognize that `_..._` containing underscores is likely a filename or identifier in italics, not malformed emphasis.

## Related

Issue #263 - Linter comparison

**Comment by joshrotenberg** [comment]
Fixed in PR #279.
